caddy/caddy.go

```diff
@@ -91,29 +78,22 @@ func (f *FrankenPHPApp) Start() error {
 		opts = append(opts, frankenphp.WithWorkers(repl.ReplaceKnown(w.FileName, ""), w.Num, w.Env, w.Watch))
 	}
 
-	_, loaded, err := phpInterpreter.LoadOrNew(mainPHPInterpreterKey, func() (caddy.Destructor, error) {
-		if err := frankenphp.Init(opts...); err != nil {
-			return nil, err
-		}
-
-		return phpInterpreterDestructor{}, nil
-	})
-	if err != nil {
+	frankenphp.Shutdown()
+	if err := frankenphp.Init(opts...); err != nil {
 		return err
 	}
 
-	if loaded {
-		frankenphp.Shutdown()
-		if err := frankenphp.Init(opts...); err != nil {
-			return err
-		}
-	}
-
 	return nil
 }
 
 func (f *FrankenPHPApp) Stop() error {
 	caddy.Log().Info("FrankenPHP stopped 🐘")
+
+	// attempt a graceful shutdown if we are not running tests
+	if flag.Lookup("test.v") == nil {
+		frankenphp.Shutdown()
```

Comment:
This cannot work because in case of reload, Caddy starts the new module before stopping the old one.

Reply:
Oh I think I get it now. There's an experimental [caddy.Exiting()](https://github.com/caddyserver/caddy/blob/e76405d55058b0a3e5ba222b44b5ef00516116aa/caddy.go#L810) api that we could maybe use instead.

Is there also a way to reload Caddy outside of tests? Running `curl "http://localhost:2019/load"` via the admin API also doesn't work on main.

Reply:
Nvm I realized you have to actually patch the whole config for a reload 

```
curl "http://localhost:2019/config/apps/frankenphp" -X PATCH --header "Content-Type: application/json" -d '{"workers"
:[{"file_name":"/go/src/app/testdata/index.php","watch":["./**/*.{php,yaml,yml,twig,env}"]}]}' -H "Cache-Control: must-revalidate"
```

This works now with the `Exiting()` check if you're fine with using an experimental api. Otherwise I'll close this PR.

Reply:
You can use `--force` if you're using [`caddy reload`](https://caddyserver.com/docs/command-line#caddy-reload) or equivalent, which is the same as passing the `Cache-Control: must-revalidate` header in the API to reload even if the config text has not changed.